### PR TITLE
Add option for whether ambiguous width characters should be fullwidth or halfwidth

### LINKED ISSUE #1: Add option to let users decide if ambiguous width characters should be 1 or 2 in width
See: https://github.com/twada/power-assert-runtime/pull/14#issuecomment-230982557

### PR DESCRIPTION
**Pull request by joelpinheiro**
Fixes #1.

Added a set of new tests to verify the correct behaviour of this feature.

### CODE REVIEW

**Comment by sindresorhus** [comment]
It should be an option in an optional options object as the second argument, not a separate method. (`stringWidth(input, [options])`). 

Reference on [how it should be implemented](https://github.com/sindresorhus/boxen/blob/b2378fae5faa3b5e4fe942a742165627b1c200a8/index.js#L65-L72) and [documented](https://github.com/sindresorhus/boxen#boxeninput-options).

File: index.js [hunk 1]
```diff
@@ -1,35 +1,226 @@
 'use strict';
 const stripAnsi = require('strip-ansi');
 const isFullwidthCodePoint = require('is-fullwidth-code-point');
+const codePointAt = require('code-point-at');
```
Comment (sindresorhus) [inline index.js]: Just use `String#codePointAt()`.

File: index.js [hunk 2]
```diff
@@ -1,35 +1,226 @@
...
 
-		if (isFullwidthCodePoint(code)) {
-			width += 2;
-		} else {
-			width++;
+		return width;
+	};
+
+	function ambiguousChar(code) {
+		if ((code === 0x00A1) ||
```
Comment (sindresorhus) [inline index.js]: What's the source of all these code point ranges?

File: test.js [hunk 3]
```diff
@@ -1,22 +1,24 @@
 import test from 'ava';
-import m from './';
+import stringWidth from './';
```
Comment (sindresorhus) [inline test.js]: Don't do unrelated changes.

File: index.js (hunk 2 above)
Comment (joelpinheiro) [inline index.js]: University of Cambridge:
[https://www.cl.cam.ac.uk/~mgk25/ucs/wcwidth.c](url)

**Comment by joelpinheiro** [comment]
Thanks for the code review @sindresorhus.

I refactor the code and I assume a default ambiguous char width of 2. Please let me know if you agree. I added a new section to documentation - API.

**Comment by sindresorhus** [comment]
Ugh, sorry @joelpinheiro. I did a review 16 days ago, but forgot to click the "Submit review" button...

**Comment by sindresorhus** [comment]
Hey, @joelpinheiro, would you be willing to finish this up? :)
